### ISSUE TITLE
Enable using patrol_cli with FVM

### ISSUE THREAD

**Issue by lyio** [leancodepl/patrol#1482]
### Steps to reproduce

1. Use `fvm` in your project
2. Do not have a global Flutter version installed
3. Install `patrol_cli`
4. Try to run `patrol test` or `patrol drive`

### Actual results

`patrol_cli` attempts to invoke `flutter` which is not available (in our case). We use `fvm` in all our projects and manage our Flutter versions like that.

It would be nice if there was a way to run Patrol in this environment without having to manage a global Flutter version.

### Logs

<details>
<summary>Logs</summary>

```
<!-- Replace this line with your logs. Do not remove the backticks! -->
```

</details>

### Patrol version

partrol: any

### Patrol Doctor output

<details>
<summary>Patrol Doctor output</summary>

```
<!-- Replace this line with your logs. Do not remove the backticks! -->
```

</details>

### Flutter Doctor output

<details>
<summary>Flutter Doctor output</summary>

```
<!-- Replace this line with your logs. Do not remove the backticks! -->
```

</details>

**Comment by bartekpacia** [comment]
Hi @lyio :)

This issue is a valid one.

The first solution that came to my mind is to have an env var `PATROL_USE_FVM`. If it is set, `patrol` will be doing `fvm flutter` and `fvm dart` internally.

Alternatively: we could automatically fallback to `fvm` if no global Flutter version is found. But I don't like such implicit behavior.

Maybe you have any other ideas?

**Comment by lyio** [comment]
While writing the issue I thought, checking for `fvm` might be a good way to handle this. Basically, your second suggestion the other way around.

I personally do not like setting a bunch of env variables, but we could combine the variable with a cli flag maybe? 
That way you could set the `PATROL_USE_FVM` or `patrol test --fvm`.

**Comment by bartekpacia** [comment]
Yeah, env var + cli flag is OK with me.

I'll try to push this issue forward on sprint planning ;P

**Comment by Jjagg** [comment]
@bartekpaci What about a `PATROL_FLUTTER_COMMAND` env variable instead? It seems more flexible and does not tie the flag and implementation to FVM specifically. Users can specify `PATROL_FLUTTER_COMMAND=fvm flutter` to use fvm or they could specify a full path to a flutter installation.

**Comment by vlazdra** [comment]
Hey @bartekpacia has this feature request made it to sprint planing? 😁

**Comment by mrocha98** [comment]
> @bartekpaci What about a `PATROL_FLUTTER_COMMAND` env variable instead? It seems more flexible and does not tie the flag and implementation to FVM specifically. Users can specify `PATROL_FLUTTER_COMMAND=fvm flutter` to use fvm or they could specify a full path to a flutter installation.

Thats a great ideia, with this env [puro](https://github.com/pingbird/puro) becomes usable too

**Comment by mvarendorff** [comment]
Hi, is there any progress on this? If not, I would like to pick this up because our projects depend on fvm and wants to adopt patrol. If that's fine with the maintainers, I would probably need a teensy bit of guidance to get going :) Let me know what you think!

**Comment by Jjagg** [comment]
@mvarendorff I did some work on this a while ago, but unfortunately lost track before finishing it up due to a job switch.

I don't remember exactly how far I got, but my work's in here:
https://github.com/Jjagg/patrol/commit/f0dfb53754cb0994914dc8bb743f2370391db03a.

**Comment by mvarendorff** [comment]
Oh that's brilliant, thanks a bunch @Jjagg! Just read through it and it looks like it's *really* far already! I'd still love a green light from the maintainers before I start; maybe @piotruela can help?

**Comment by piotruela** [comment]
Hi @mvarendorff, we will gladly add this to patrol, so if you're willing, feel free. Also thanks @Jjagg for the initial work. Let's do it so it will support other commands than just `fvm flutter`. As far as I see this is how it is done at this point by @Jjagg.

**Comment by mvarendorff** [comment]
For those following the thread, this has been released with `patrol_cli@2.7.0` earlier today! A colleague of mine already gave it a testride and he's happy!

**Comment by piotruela** [comment]
Closing this as this has been released in patrol_cli 2.7.0. Thanks @mvarendorff for your contribution!
